Libraries/LibURL/Pattern/PatternParser.cpp

```diff
@@ -0,0 +1,405 @@
+
+        // 6. If fixed token is null, then set fixed token to the result of running try to consume a token given
+        //     parser and "escaped-char".
+        if (!fixed_token)
+            fixed_token = parser.try_to_consume_a_token(Token::Type::EscapedChar);
+
+        // 7. If fixed token is not null:
+        if (fixed_token) {
+            // 1. Append fixed token’s value to parser’s pending fixed value.
+            parser.m_pending_fixed_value.append(fixed_token->value);
+
+            // 2. Continue.
+            continue;
+        }
+
+        // 8. Let open token be the result of running try to consume a token given parser and "open".
+        auto const* open_token = parser.try_to_consume_a_token(Token::Type::Open);
+
+        // 9. If open token is not null:
+        if (open_token) {
+            // 1. Let prefix be the result of running consume text given parser.
+            auto prefix = parser.consume_text();
+
+            // 2. Set name token to the result of running try to consume a token given parser and "name".
+            name_token = parser.try_to_consume_a_token(Token::Type::Name);
+
+            // 3. Set regexp or wildcard token to the result of running try to consume a regexp or wildcard token
+            //    given parser and name token.
+            regexp_or_wildcard_token = parser.try_to_consume_a_regexp_or_wildcard_token(name_token);
+
+            // 4. Let suffix be the result of running consume text given parser.
+            auto suffix = parser.consume_text();
+
+            // 5. Run consume a required token given parser and "close".
+            TRY(parser.consume_a_required_token(Token::Type::Close));
+
+            // 6. Let modifier token to the result of running try to consume a modifier token given parser.
+            auto const* modifier_token = parser.try_to_consume_a_modifier_token();
+
+            // 7. Run add a part given parser, prefix, name token, regexp or wildcard token, suffix, and modifier token.
+            TRY(parser.add_a_part(prefix, name_token, regexp_or_wildcard_token, suffix, modifier_token));
+
+            // 8. Continue.
+            continue;
+        }
+
+        // 10. Run maybe add a part from the pending fixed value given parser.
+        TRY(parser.maybe_add_a_part_from_the_pending_fixed_value());
+
+        // 11. Run consume a required token given parser and "end".
+        TRY(parser.consume_a_required_token(Token::Type::End));
+    }
+
+    if constexpr (URL_PATTERN_DEBUG) {
+        dbgln("Pattern parser produced the part list:");
+        for (auto const& part : parser.m_part_list) {
+            dbgln("Type {}, Value '{}', Modifier {}, Name '{}', Prefix '{}', Suffix '{}'",
+                Part::type_to_string(part.type), part.value, Part::convert_modifier_to_string(part.modifier),
+                part.name, part.prefix, part.suffix);
+        }
+    }
+    // 4. Return parser’s part list.
+    return parser.m_part_list;
```

Comment:
If we're done with `m_part_list` at this point, we could `move` it, as RVO doesn't apply here
```c++
return move(parser.m_part_list)`;
```